### PR TITLE
🌐 Add Japanese translation for `docs/ja/docs/advanced/nosql-databases.md`

### PR DESCRIPTION
**Pull request by sUeharaE4**
Relates to #1572

This PR translates advanced/nosql-databases.md.

I left "databases" in the page title as it is because there is no word for the plural of database in Japanese.  
This is my first pull request, so I apologize if there are any problems. I would appreciate it if you could check it.

### CODE REVIEW

**Review by dimaqq — APPROVED** [review]
I'm not a native speaker, but this seems solid!

**Review by wakabame — APPROVED** [review]
Excellent work!!

**Comment by dimaqq** [comment]
@tiangolo this should be ready, let's merge!

**Review by tokusumi — CHANGES_REQUESTED** [review]
Thank you for great work!!!
I'd like to suggests the followings, although already you've gotten enough approve.

File: docs/ja/docs/advanced/nosql-databases.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+他にもこれらのNoSQLデータベースを利用することが出来ます:
+
+* **MongoDB**
+* **Cassandra**
+* **CouchDB**
+* **ArangoDB**
+* **ElasticSearch** など。
+
+!!! 豆知識
```
Comment (tokusumi) [inline docs/ja/docs/advanced/nosql-databases.md:15]:
```suggestion
!!! tip "豆知識"
```
I'm not sure but maybe this is (or was) right syntax.

File: docs/ja/docs/advanced/nosql-databases.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
+# NoSQL (分散 / ビッグデータ) Databases
+
+**FastAPI** はあらゆる <abbr title="分散データベース (Big Data), や 'Not Only SQL'">NoSQL</abbr>と統合することもできます。
```
[the remark this answers is not in the document]
Comment (sUeharaE4) [reply docs/ja/docs/advanced/nosql-databases.md]: Thank you to suggests. I applied at here.
https://github.com/tiangolo/fastapi/pull/4205/commits/6866770bda9532a4860e7c6e661ab7efcba6a60f#diff-101eb49c79208a3498543c29b9170aaabf8ce9c8dbdc16b3eb63f4010c9081eeL3

File: docs/ja/docs/advanced/nosql-databases.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+リレーショナルデータベースの世界でいう"database"(データベースサーバではなく特定のdatabase)と類似しています。
+
+**MongoDB** で例えると"collection"と似た概念です。
+
+次のコードでは主に `Bucket` を利用してCouchbaseを操作します。
+
+この関数では:
+
+* **Couchbase** クラスタ(1台の場合もあるでしょう)に接続し
```
[the remark this answers is not in the document]
Comment (sUeharaE4) [reply docs/ja/docs/advanced/nosql-databases.md]: Thank you to suggests. I applied arround here.
https://github.com/tiangolo/fastapi/pull/4205/commits/6866770bda9532a4860e7c6e661ab7efcba6a60f#diff-101eb49c79208a3498543c29b9170aaabf8ce9c8dbdc16b3eb63f4010c9081eeL48

File: docs/ja/docs/advanced/nosql-databases.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    * タイムアウトのデフォルト値を設定して
+* 作成した`Bucket`インスタンスを返却します。
+
+```Python hl_lines="12-21"
+{!../../../docs_src/nosql_databases/tutorial001.py!}
+```
+
+## Pydantic モデルの作成
+
+**Couchbase**のdocumentは実際には単にJSONオブジェクトなのでPydanticを利用してモデルにすることが出来ます。
```
[the remark this answers is not in the document]
Comment (sUeharaE4) [reply docs/ja/docs/advanced/nosql-databases.md]: Thank you to suggests. I applied at here.
https://github.com/tiangolo/fastapi/pull/4205/commits/6866770bda9532a4860e7c6e661ab7efcba6a60f#diff-101eb49c79208a3498543c29b9170aaabf8ce9c8dbdc16b3eb63f4010c9081eeL63

File: docs/ja/docs/advanced/nosql-databases.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+### `User` モデル
+
+まずは`User`モデルを作成してみましょう:
+
+```Python hl_lines="24-28"
+{!../../../docs_src/nosql_databases/tutorial001.py!}
+```
+
+このモデルは*path operation*で利用できる関数で使用するので`hashed_password`はモデルに含めません。
```
[the remark this answers is not in the document]
Comment (sUeharaE4) [reply docs/ja/docs/advanced/nosql-databases.md]: Thank you to suggests. I applied at here.
https://github.com/tiangolo/fastapi/pull/4205/commits/6866770bda9532a4860e7c6e661ab7efcba6a60f#diff-101eb49c79208a3498543c29b9170aaabf8ce9c8dbdc16b3eb63f4010c9081eeL73

File: docs/ja/docs/advanced/nosql-databases.md [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+こちらは実際にデータベースに保存されるデータを保持します。
+
+`User`モデルの持つ全ての属性に加えていくつかの属性を追加するのでPydanticの`BaseModel`を継承せずに`User`のサブクラスとして定義します:
+
+```Python hl_lines="31-33"
+{!../../../docs_src/nosql_databases/tutorial001.py!}
+```
+
+!!! note
```
[the remark this answers is not in the document]
Comment (sUeharaE4) [reply docs/ja/docs/advanced/nosql-databases.md]: Thank you to suggests. I applied at here.
https://github.com/tiangolo/fastapi/pull/4205/commits/6866770bda9532a4860e7c6e661ab7efcba6a60f#diff-101eb49c79208a3498543c29b9170aaabf8ce9c8dbdc16b3eb63f4010c9081eeL87

File: docs/ja/docs/advanced/nosql-databases.md [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    データベースに保存される`hashed_password`と`type`フィールドを`UserInDB`モデルに保持させていることに注意してください。
+
+    しかしこれらは(*path operation*で返却する)一般的な`User`モデルには含まれません
+
+## user の取得
+
+それでは次の関数を作成しましょう:
+
+* username を取得して
```
[the remark this answers is not in the document]
Comment (sUeharaE4) [reply docs/ja/docs/advanced/nosql-databases.md]: Thank you to suggests. I applied at here.
https://github.com/tiangolo/fastapi/pull/4205/commits/6866770bda9532a4860e7c6e661ab7efcba6a60f#diff-101eb49c79208a3498543c29b9170aaabf8ce9c8dbdc16b3eb63f4010c9081eeL87

File: docs/ja/docs/advanced/nosql-databases.md [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+* 作成したIDでdocumentを取得し
+* documentの内容を`UserInDB`モデルに設定する
+
+*path operation関数*とは別に、`username`(またはその他のパラメータ)からuserを取得することだけに特化した関数を作成することで、より簡単に複数の部分で再利用したり<abbr title="コードで書かれた自動テストで、他のコードが正しく動作しているかどうかをチェックするもの。">ユニットテスト</abbr>を追加することができます。
+
+```Python hl_lines="36-42"
+{!../../../docs_src/nosql_databases/tutorial001.py!}
+```
+
+### f-strings
```
[the remark this answers is not in the document]
Comment (sUeharaE4) [reply docs/ja/docs/advanced/nosql-databases.md:107]: Thank you to suggests. I applied at here.
https://github.com/tiangolo/fastapi/pull/4205/commits/6866770bda9532a4860e7c6e661ab7efcba6a60f#diff-101eb49c79208a3498543c29b9170aaabf8ce9c8dbdc16b3eb63f4010c9081eeL96

File: docs/ja/docs/advanced/nosql-databases.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+まずはImportしましょう。今はその他のソースコードに注意を払う必要はありません。
+
+```Python hl_lines="3-5"
+{!../../../docs_src/nosql_databases/tutorial001.py!}
+```
+
+## "document type" として利用する定数の定義 
+
+documentで利用する固定のフィールド`type`を用意しておきます。
```
[the remark this answers is not in the document]
Comment (sUeharaE4) [reply docs/ja/docs/advanced/nosql-databases.md]: Thank you to suggests. I applied at here.
https://github.com/tiangolo/fastapi/pull/4205/commits/6866770bda9532a4860e7c6e661ab7efcba6a60f#diff-101eb49c79208a3498543c29b9170aaabf8ce9c8dbdc16b3eb63f4010c9081eeL28

File: docs/ja/docs/advanced/nosql-databases.md (hunk 1 above)
Comment (sUeharaE4) [reply docs/ja/docs/advanced/nosql-databases.md]: Thank you to suggests. I applied at here.
https://github.com/tiangolo/fastapi/pull/4205/commits/6866770bda9532a4860e7c6e661ab7efcba6a60f#diff-101eb49c79208a3498543c29b9170aaabf8ce9c8dbdc16b3eb63f4010c9081eeL15

File: docs/ja/docs/advanced/nosql-databases.md [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+
+他にもこれらのNoSQLデータベースを利用することが出来ます:
+
+* **MongoDB**
+* **Cassandra**
+* **CouchDB**
+* **ArangoDB**
+* **ElasticSearch** など。
+
+!!! tips "豆知識"
```
Comment (tokusumi) [inline docs/ja/docs/advanced/nosql-databases.md:15]:
```suggestion
!!! tip "豆知識"
```
typo.

You can cofirm right style (green color and fire icon) for "tips (豆知識)" admonition block as:
- [source](https://github.com/tiangolo/fastapi/blob/master/docs/ja/docs/deployment/versions.md#%E3%83%90%E3%83%BC%E3%82%B8%E3%83%A7%E3%83%B3%E3%81%AB%E3%81%A4%E3%81%84%E3%81%A6):
- [preview](https://github.com/tiangolo/fastapi/blob/master/docs/ja/docs/deployment/versions.md#%E3%83%90%E3%83%BC%E3%82%B8%E3%83%A7%E3%83%B3%E3%81%AB%E3%81%A4%E3%81%84%E3%81%A6)

**Review by tokusumi — CHANGES_REQUESTED** [review]

File: docs/ja/docs/advanced/nosql-databases.md (hunk 10 above)
Comment (sUeharaE4) [reply docs/ja/docs/advanced/nosql-databases.md]: I'm sorry! I applied this fix. Thank you for your quickly and carefully review.

**Review by tokusumi — APPROVED** [review]
LGTM

**Comment by tiangolo** [comment]
Thanks for the contribution @sUeharaE4! :cake: 

And thank you for the reviews @dimaqq, @wakabame, and @tokusumi! :coffee:

**Pull request merged**